pkg/handler/handler.go

```diff
@@ -94,7 +95,7 @@ type patchOperation struct {
 	Value interface{} `json:"value,omitempty"`
 }
 
-func addEnvToContainer(container *corev1.Container, mountPath, tokenName, volName, roleName string) {
+func addEnvToContainer(container *corev1.Container, mountPath, hostTokenFilePath, volName, roleName string) {
```

Comment:
@micahhausler we'll need to pass both `mountPath` and `hostTokenFilePath`. as the `mountPath` will only refer to the directory containing the token file and not the file itself. 

Also, I'm not converting `mountPath` to a platform-specific path as kubernetes supports specifying unix-style file paths for windows container workload volume mounts.

Let me know if I should change this to platform-specific path.

Reply:
naming nit: can you remove `host`? Its not actually a host path (same for the `updatePodSpec` var name)